app/models/match.rb

```diff
@@ -0,0 +1,38 @@
+class Match < ActiveRecord::Base
+  validates :given_swipe, :received_swipe, presence: true
+
+  belongs_to :given_swipe, foreign_key: :given_swipe_id, class_name: "Swipe"
+  belongs_to :received_swipe, foreign_key: :received_swipe_id, class_name: "Swipe"
+
+  has_many :messages
+
+  delegate :my_item, to: :given_swipe
+  delegate :other_item, to: :given_swipe
+
+  alias item_1 my_item
+  alias item_2 other_item
+
+  def other(item)
+    if item == self.item_1
```

Comment:
This could be a ternary (a simple if this then that else other is about the only time they are preferable) so
 ```
item == item_1 ? item_2 : item_1
``` 
The self. is not necessary if you're calling an instance method (only when writing with an attr_accessor.)